### ISSUE TITLE
Set ceil_mode=True cannot solve the shape error.

### ISSUE THREAD

**Issue by Flames60** [MTlab/onnx2caffe#10]
Hi dude, you said this in another issue:
`Hi, this is caused by the difference of pooling between caffe and pytorch. Caffe's pooling use "ceil" mode while pytorch's use "floor" mode by default. When the output shapes are not integers, they behaves different. For example, the output of Pooling(2, 2)(1, 3, 129, 129) will be (1, 3, 65, 65) in caffe and (1, 3, 64, 64) in pytorch. Unfortunately, there are several such kind of pooling in pytorch's offical resnet model. You need to set ceil_mode=True in pytorch when converting this kind of pooling to caffe. Check #3 for detail.`
But when I tried this ,I cannot convert the model from pytorch to onnx with torch0.4.0, it said this:
`ONNX export failed: Couldn't export operator aten::avg_pool2d`
Although I can do it with torch0.4.1 and torch1.0, this project cannot support the operator 'ATEN' and 'GlobalAvgPool'.
I am trying to add the  'GlobalAvgPool' to your project, or do you have a better approach?
Thanks a lot!

**Comment by YukiNagato** [comment]
I think the best way is to change the pooling params. I change "self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)" to "self.maxpool = nn.MaxPool2d(kernel_size=2, stride=2)", and  it's runnable. But you may have to finetune or pretrain your network. The runnable resnet50 can be found in model_generator/resnet50.py.

**Comment by Flames60** [comment]
> I think the best way is to change the pooling params. I change "self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)" to "self.maxpool = nn.MaxPool2d(kernel_size=2, stride=2)", and it's runnable. But you may have to finetune or pretrain your network. The runnable resnet50 can be found in model_generator/resnet50.py.

Yes, I can convert the pretrained resnet50 in this way, but I have changed the resnet18(myResnet18), which has a global average pooling layer(GAP) implemented with the function AdaptiveAvgPool, now I am stucked on converting this.
If I add 1 on the kernel size, such that the original size is (24,8), and now is (25,9), the output in torch will be (0,0) instead of (1,1), which is not I want after a GAP, can you help me?

**Comment by Flames60** [comment]
> I think the best way is to change the pooling params. I change "self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)" to "self.maxpool = nn.MaxPool2d(kernel_size=2, stride=2)", and it's runnable. But you may have to finetune or pretrain your network. The runnable resnet50 can be found in model_generator/resnet50.py.

Thank you! I have tried your ways to convert resnet50 few days ago, like another issue said, and it works! But it is not about the maxpool this time, I changed the resnet and implemented the GAP on this way to solve the problem
`class MyAdaptiveAvgPool2d(nn.Module):
    def __init__(self, sz=None):
        super(MyAdaptiveAvgPool2d,self).__init__()


    def forward(self, x):
        inp_size = x.size()
        m = torch.nn.AvgPool2d( #input=x,
                  kernel_size= (inp_size[2], inp_size[3]))
        print(inp_size[2:])
        return m(x)
`
Good luck!
